api/environments/identities/traits/views.py

```diff
@@ -145,21 +88,16 @@ def partial_update(self, request, *args, **kwargs):
         ]
     )
     def destroy(self, request, *args, **kwargs):
-        delete_all_traits = request.query_params.get("deleteAllMatchingTraits")
-        if delete_all_traits and delete_all_traits in ("true", "True"):
+        if request.query_params.get("deleteAllMatchingTraits") in ("true", "True"):
             trait = self.get_object()
-            self._delete_all_traits_matching_key(
-                trait.trait_key, trait.identity.environment
-            )
+            Trait.objects.filter(
```

Comment:
@matthewelwell  I don't think we are planning to support this for edge identities?

Reply:
Yeah, that's ok - I don't think it's used tbh.